### DISCUSSION TITLE
Is there support for compression of jsonified pydantic objects sent in responses?

### DISCUSSION THREAD

**Discussion by zachglee** [fastapi/fastapi#6094] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
n/a
```

### Description

I have an application where I return large json documents. I would like to compress them. Does FastAPI currently do this for me, and if not, is there a way I can enable compression? My understanding is that if I have full control of the http response I can do this by compressing myself and specifying certain headers (https://developer.mozilla.org/en-US/docs/Web/HTTP/Compression) but I am wondering if there's a way for me to avoid that and be able to keep using FastAPI's nice system of being able to return a pydantic object and have it be automatically serialized?

Thanks!

### Operating System

macOS

### FastAPI Version

0.77.1

### Python Version

3.9.14

**Comment by jgould22** [comment, marked as answer]
Is this what you are looking for ?

https://fastapi.tiangolo.com/advanced/middleware/#gzipmiddleware

**Comment by zachglee** [comment]
> Is this what you are looking for ?
> 
> https://fastapi.tiangolo.com/advanced/middleware/#gzipmiddleware

Yes, thank you!!
